include/zephyr/drivers/emul_sensor.h

```diff
@@ -26,11 +26,18 @@
  */
 __subsystem struct emul_sensor_backend_api {
 	/** Sets a given fractional value for a given sensor channel. */
-	int (*set_channel)(const struct emul *target, enum sensor_channel ch, q31_t value,
+	int (*set_channel)(const struct emul *target, enum sensor_channel ch, const q31_t *value,
 			   int8_t shift);
 	/** Retrieve a range of sensor values to use with test. */
 	int (*get_sample_range)(const struct emul *target, enum sensor_channel ch, q31_t *lower,
 				q31_t *upper, q31_t *epsilon, int8_t *shift);
+	/** Set the offset of a given chanel. */
+	int (*set_offset)(const struct emul *target, enum sensor_channel ch, q31_t *offsets,
+			  int8_t shift);
```

Comment:
Attributes currently use the old representation of sensor values which are a pain point for many drivers. I purposefully don't want to add more APIs that use `sensor_value` as I believe it will be deprecated in the near future. Reading data will be using the mapping we've introduced with the decoder of channel -> data type. Attributes will likely follow suit. As such, I think it's best to have explicit functions which can provide these better data type bindings to the attributes. It's possible that once we land on the right attribute API, we'll have to collapse these into 1 function, but for now I would prefer not to keep building on the attribute API.

Reply:
I'm perfectly happy to dump `sensor_value` (we should be using Q31 exclusively in the backend API) or making it a generic pointer to allow for flexibility, but I disagree with creating separate API functions for each kind of attribute. Having a single one (1) reduces code duplication, (2) avoids needing to keep adding more functions when somebody wants the generic test to cover a new attribute, and (3) makes it more awkward for the test to discover supported attributes --with this solution the test would need to test-call each attribute setter function to discover capabilities (and also have a switch block in the test body for fanning out each type of attribute to the appropriate setter)

Reply:
Ah, I see. I think this goes beyond just q31_t. I'm getting at the fact that some channel/attribute combinations are bools, some are 1D `q31_t` some are 3D `q31_t` some might be be counters like `uint64_t`. Until we're ready to tackle the attribute API, I don't want to keep building on top of it. Lets assume there's a unified backend API that always accepted `q31_t *values, size_t value_count` as arguments. If an attribute requires a `bool` how would you represent it?

I think it's easier to have separate functions that have the explicit value counts and types as part of the signature. Then when we clean up the attribute API it'll be easy to merge them if/when needed.

Reply:
Done